RATapi/utils/plotting.py

```diff
@@ -241,10 +242,11 @@ def plot_ref_sld(
     """
     data = PlotEventData()
 
+    # We need to take a copy of results and SLD in case the data to plot is modified later
```

Comment:
Do you have a scenario in mind for this? Or is one of our plots modifying the data?

Reply:
Yup, a few lines below this we change `data.reflectivity` and `data.sld` if we are plotted a shaded plot - this is the switch of data that this PR is for. I'll change the comment to clarify.